src/bc_protocol.rs

```diff
@@ -242,7 +242,7 @@ impl BcCamera {
         Ok(())
     }
 
-    pub fn start_video(&self, data_out: &mut dyn Write) -> Result<Never> {
+    pub fn start_video(&self, data_out: &mut dyn Write, stream: &String) -> Result<Never> {
```

Comment:
This should be `&str`, a reference to a string slice.  Can you rename it to `stream_name`? (Nitpick I know)

Reply:
This should be done now